### PR TITLE
AIMS-246: User access

### PR DESCRIPTION
**Pull request by jgondron**
Why: Need to restrict access to the application to a specified list of users
How: Added an admin field to the users table and modified the ApplicationController to check this. Essentially only administrators of the application will have any access, and they will all have access to a form to edit the users. In order to give permission to the first user you can configure an admin_user_name in the config.

### CODE REVIEW

**Comment by jaronkk** [comment]
Can you add a spec asserting that you receive a 404 if you're logged in as a user with admin set to false?

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -10,4 +11,20 @@ class ApplicationController < ActionController::Base
   def redirect_to_sign_in
     redirect_to new_user_session_path
   end
+
+  def redirect_to_unauthorized
+    raise_404("User not an admin")
+  end
+
```
Comment (rfox2) [inline app/controllers/application_controller.rb]: Should this be a 401?

File: config/environments/development.rb [hunk 2]
```diff
@@ -41,6 +41,9 @@
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Configures a user as an administrator
+  config.admin_user_name = "jgondron"
```
Comment (rfox2) [inline config/environments/development.rb]: Good - none of the rest of us want this massive responsibility - you put this in pprd and prod too right?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jgondron) [inline app/controllers/application_controller.rb]: I'm following DEC, we throw 404

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def update
+    user_id = params["user_id"]
```
Comment (rfox2) [inline app/controllers/users_controller.rb]: Could we add a basic feature test for this?

File: config/environments/development.rb (hunk 2 above)
Comment (jgondron) [inline config/environments/development.rb]: I can!

**Comment by jaronkk** [comment]
I added an example asserting that it redirects to the sign in path if you're not logged in.

**Comment by jaronkk** [comment]
:+1:

**Comment by rfox2** [comment]
:+1: Let's put "mvanneve" as the admin for prod and pprd, and @jgondron as the admin in staging

**Pull request merged**